LogicBuilder.Expressions.Utils/Expansions/ExpansionsHelper.cs

```diff
@@ -175,6 +175,13 @@ private static Expression BuildSelectorExpression(Expression sourceExpression, L
                 }
                 else
                 {
+                    if (parent.Type != parts[i].ParentType)
```

Comment:
I think we want to throw an exception here if `parent.Type` is not assignable form `parts[i].ParentType` ?

Reply:
I assumed that OData would take care of this check beforehand, and there was no tests for the validity of getting the `parts[i].MemberName` member out of the `parent.Type` type (which caused my issue btw).

Both of these errors return perfectly readable and troubleshootable error messages so.. What do you think ?

Reply:
Just FYI this library does not assume an OData caller. Not a huge deal.  A unique message makes it easier to pinpoint even `throw new InvalidOperationException("{EFF2398E-8BA7-4A52-A098-AEDF6280F790}");` is helpful.